docs/src/main/asciidoc/resource-handling.adoc

```diff
@@ -103,9 +109,65 @@ spring:
           endpoint:  ${STORAGE_FILESHARE_ENDPOINT}
 ----
 
+==== Autowire a resource
+You can use the annotation of `@Value("azure-blob://[your-container-name]/[your-blob-name]")` to autowire a blob Resource.
+----
+@Value("azure-blob://[your-container-name]/[your-blob-name]")
+private Resource storageResource;
+----
+
+You can use the annotation of @Value("azure-file://[your-fileshare-name]/[your-file-name]") to autowire a file Resource.
+
+----
+@Value("azure-file://[your-fileshare-name]/[your-file-name]")
+private Resource storageResource;
+----
+
+==== Read and write to a resource
+
+You can read a resource from Azure Blob storage with `getInputStream()` method.
+
+----
+ this.storageResource.getInputStream();
+----
+
+You can write to a resource in Azure Blob storage by casting the Spring `Resource` to `WritableResource`.
+
+----
+ ((WritableResource) this.azureBlobResource).getOutputStream();
+----
+
+
+#### Multipart upload
+Files larger than 4 MiB will be uploaded to Azure Storage in parallel.
 
 === Samples
 
-Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0[azure-spring-boot-samples] for more details.
+Please refer to link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_4.0-beta.2/storage[azure-spring-boot-samples] for more details.
+
 
+==== Other operations
+The Spring Resource abstraction for Azure Storage also supports link:https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#resources[other operations] defined in Spring's `Resource` and `WritableResource` interface.
+
+==== Autowire the BlobServiceClientBuilder
+You can autowire the `BlobServiceClientBuilder` and create a client using:
+----
+@Autowire
+private BlobServiceClientBuilder blobServiceClientBuilder;
 
+private final BlobServiceAsyncClient blobServiceAsyncClient = blobServiceClientBuilder.buildAsyncClient();
+
+----
+==== Search for resources
+You can use implementation class `AzureStorageBlobProtocolResolver` of `ResourcePatternResolver` to search `blob` resource, and `AzureStorageFileProtocolResolver` of `ResourcePatternResolver` to search `file` resource
+
+* Pattern search, the **searchPattern** should start with `azure-blob://` or `azure-file://`. Such as `azure-blob://*/*`, it means list all blobs in all containers; `azure-blob://demo-container/**`, it means list all blobs in the demo-container container, including any sub-folder.
+* Location search, the **searchLocation** should start with `azure-blob://` or `azure-file://`, the remaining file path should exist, otherwise an exception will be thrown.
+
+----
+@Autowire
+ResourcePatternResolver storageResourcePatternResolver;
+
+Resource[] resources = storageResourcePatternResolver.getResources(searchPattern);
+Resource resource = storageResourcePatternResolver.getResource(searchLocation);
```

Comment:
Should we put this part under the `basic usages` section? These are the champion scenarios we were talking about.

Reply:
I think so